src/panels/config/tags/ha-config-tags.ts

```diff
@@ -0,0 +1,293 @@
+            "ui.panel.config.tags.headers.last_scanned"
+          ),
+          sortable: true,
+          direction: "desc",
+          width: "20%",
+          template: (last_scanned_datetime) => html`
+            ${last_scanned_datetime
+              ? html`<ha-relative-time
+                  .hass=${this.hass}
+                  .datetimeObj=${last_scanned_datetime}
+                ></ha-relative-time>`
+              : this.hass.localize("ui.components.relative_time.never")}
+          `,
+        };
+      }
+      if (canWriteTags) {
+        columns.write = {
+          title: "",
+          type: "icon-button",
+          template: (_write, tag: any) => html` <mwc-icon-button
+            .tag=${tag}
+            @click=${(ev: Event) =>
+              this._openWrite((ev.currentTarget as any).tag)}
+            title=${this.hass.localize("ui.panel.config.tags.write")}
+          >
+            <ha-svg-icon .path=${mdiContentDuplicate}></ha-svg-icon>
+          </mwc-icon-button>`,
+        };
+      }
+      columns.automation = {
+        title: "",
+        type: "icon-button",
+        template: (_automation, tag: any) => html` <mwc-icon-button
+          .tag=${tag}
+          @click=${(ev: Event) =>
+            this._createAutomation((ev.currentTarget as any).tag)}
+          title=${this.hass.localize("ui.panel.config.tags.create_automation")}
+        >
+          <ha-svg-icon .path=${mdiRobot}></ha-svg-icon>
+        </mwc-icon-button>`,
+      };
+      columns.edit = {
+        title: "",
+        type: "icon-button",
+        template: (_settings, tag: any) => html` <mwc-icon-button
+          .tag=${tag}
+          @click=${(ev: Event) =>
+            this._openDialog((ev.currentTarget as any).tag)}
+          title=${this.hass.localize("ui.panel.config.tags.edit")}
+        >
+          <ha-svg-icon .path=${mdiCog}></ha-svg-icon>
+        </mwc-icon-button>`,
+      };
+      return columns;
+    }
+  );
+
+  private _data = memoizeOne((tags: Tag[]): TagRowData[] => {
+    return tags.map((tag) => {
+      return {
+        ...tag,
+        display_name: tag.name || tag.id,
+        last_scanned_datetime: tag.last_scanned
+          ? new Date(tag.last_scanned)
+          : null,
+      };
+    });
+  });
+
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.firstUpdated(changedProperties);
+    this._fetchTags();
+    if (this.hass && this.hass.auth.external) {
+      getExternalConfig(this.hass.auth.external).then((conf) => {
+        this._canWriteTags = conf.canWriteTag;
+      });
+    }
+  }
+
+  protected hassSubscribe() {
+    return [
+      this.hass.connection.subscribeEvents<TagScannedEvent>((ev) => {
+        const foundTag = this._tags.find((tag) => tag.id === ev.data.tag_id);
+        if (!foundTag) {
+          this._fetchTags();
+          return;
+        }
+        foundTag.last_scanned = ev.time_fired;
+        this._tags = [...this._tags];
```

Comment:
Why are we copying it and not just requestUpdate ?

Reply:
Because of memoize